plugins/secretstores/gdchauth/gdchauth.go

```diff
@@ -0,0 +1,276 @@
+	serviceAccountTokenType = "urn:k8s:params:oauth:token-type:serviceaccount"
+)
+
+func (*GdchAuth) SampleConfig() string {
+	return sampleConfig
+}
+
+// GdchAuth is the main authenticator struct
+type GdchAuth struct {
+	ServiceAccountFile string          `toml:"service_account_file"`
+	Audience           string          `toml:"audience"`
+	TokenExpiryBuffer  config.Duration `toml:"token_expiry_buffer"`
+	tls.ClientConfig   `toml:"tls"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	token       string
+	tokenExpiry time.Time
+	tokenMutex  sync.RWMutex
+	saKey       *ServiceAccountKey
+	httpClient  *http.Client
+}
+
+type ServiceAccountKey struct {
+	PrivateKeyID        string `json:"private_key_id"`
+	PrivateKey          string `json:"private_key"`
+	ServiceIdentityName string `json:"name"`
+	TokenURI            string `json:"token_uri"`
+	Project             string `json:"project"`
+
+	parsedKey     crypto.Signer
+	signingMethod jwt.SigningMethod
+}
+
+func (g *GdchAuth) Init() error {
+	if g.ServiceAccountFile == "" {
+		return errors.New("service_account_file is required")
+	}
+	if g.TokenExpiryBuffer == 0 {
+		g.TokenExpiryBuffer = config.Duration(5 * time.Minute)
+	}
+
+	keyData, err := os.ReadFile(g.ServiceAccountFile)
+	if err != nil {
+		return fmt.Errorf("failed to read service account file: %w", err)
+	}
+
+	g.saKey = &ServiceAccountKey{}
+	if err := json.Unmarshal(keyData, g.saKey); err != nil {
+		return fmt.Errorf("failed to parse service account JSON: %w", err)
+	}
+
+	if err := g.parsePrivateKey(); err != nil {
+		return err
+	}
+
+	return g.buildHTTPClient()
+}
+
+// Get retrieves the token. The key is ignored as this secret store only provides one secret.
+func (g *GdchAuth) Get(_ string) ([]byte, error) {
+	token, err := g.GetToken(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return []byte(token), nil
+}
+
+// List returns the list of secrets provided by this store.
+func (*GdchAuth) List() ([]string, error) {
+	return []string{"token"}, nil
+}
+
+// Set is not supported for the gdchauth secret store.
+func (*GdchAuth) Set(_, _ string) error {
+	return errors.New("setting secrets is not supported")
+}
+
+// GetResolver returns a resolver function for the secret.
+func (g *GdchAuth) GetResolver(key string) (telegraf.ResolveFunc, error) {
+	return func() ([]byte, bool, error) {
+		s, err := g.Get(key)
+		return s, true, err
+	}, nil
+}
+
+// SetLogger sets the logger for the authenticator.
+func (g *GdchAuth) SetLogger(log telegraf.Logger) {
+	g.Log = log
+}
+
+// GetToken retrieves a GDCH auth token. It caches the token and reuses it
+// until it is within the 'token_expiry_buffer' of its expiry time.
+func (g *GdchAuth) GetToken(ctx context.Context) (string, error) {
+	g.tokenMutex.RLock()
+
+	if g.token != "" && time.Now().Before(g.tokenExpiry.Add(-time.Duration(g.TokenExpiryBuffer))) {
+		g.tokenMutex.RUnlock()
+		g.Log.Debug("Using cached auth token")
```

Comment:
Do we really need this? If so, please use the trace level here and also for other "developer related" outputs...
```suggestion
		g.Log.Trace("Using cached auth token")
```